### PR TITLE
Fixed the list to display entries alphabetically

### PR DESCRIPTION
**Pull request by Perseus**
Fixes issue #966 

Removed the conditional check for `$list_id` hence displaying the list items, which are retrieved alphabetically from the database, in order.

### CODE REVIEW

**Comment by Perseus** [comment]
I'm also working on the type-ahead selector. Wanted to know how you guys wanted it done. A separate text field which the user can type in to find a list or something else?

**Comment by Perseus** [comment]
@tmccormi

**Comment by teryhill** [comment]
Fails testing.

**Comment by Perseus** [comment]
Is the top option still "language"? What is the issue?

**Comment by teryhill** [comment]
With the change you get this

image

The option changes in the drop down but the listed items do not change. Then if you select a list you get this.

image

Which shows the list but the wrong header.

**Comment by teryhill** [comment]
This is what I came up with.

image

Then when you select a list you get the list and the save button.

image

**Comment by Perseus** [comment]
@teryhill I changed it to be blank initially and nothing displays and then can be selected as needed. 
2438bc9e03fc3f0e1ce0f864c8ccc03f
dbed3daef352c64399db5a38f4f83f67
f6438d0248f62a2a29c6e32cbae7dbfb
Does this look fine?

**Comment by teryhill** [comment]
Yes it does look fine.

File: interface/super/edit_list.php [hunk 1]
```diff
@@ -858,7 +861,7 @@ function mysubmit() {
 </p>
 
 <center>
-
+<?php if ($list_id != ' '): ?>
```
Comment (teryhill) [inline interface/super/edit_list.php]: Should use {

File: interface/super/edit_list.php [hunk 2]
```diff
@@ -1046,7 +1049,7 @@ function mysubmit() {
 </center>
 
 </form>
-
+<?php endif; ?>
```
Comment (teryhill) [inline interface/super/edit_list.php]: Should use }

File: interface/super/edit_list.php (hunk 1 above)
Comment (Perseus) [reply interface/super/edit_list.php]: Alright I'll fix that.

**Comment by aethelwulffe** [comment]
OK, I just went and tested an out-of-date commit...Lemme try again.

**Comment by Perseus** [comment]
@aethelwulffe Are there any future plans to upgrade the jQuery versions being used? The edit list page uses 1.2 and most of the libraries having type-ahead selectors don't really support that. I could work on something just for that but it'd be easier if jQuery was updated instead. Less legacy code to maintain

**Comment by teryhill** [comment]
@Perseus it is an on going project. Issue 416 has the info on it.

**Comment by Perseus** [comment]
Please review the commit @teryhill @aethelwulffe . I couldn't find any errors with the updated jquery version in the edit_list page and the list seems to work properly. Please let me know if any changes are needed.

File: interface/super/edit_list.php [hunk 3]
```diff
@@ -618,9 +618,11 @@ function writeITLine($it_array) {
 <?php html_header_show();?>
 
 <!-- supporting javascript code -->
-<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/js/jquery.js"></script>
-
+<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/js/jquery-3.3.1.min.js"></script>
```
Comment (aethelwulffe) [inline interface/super/edit_list.php]: `require_once $GLOBALS['srcdir'].'/headers.inc.php'; ` is on line 22 of this file.  

To call a single library from the assets/js/ directory, use `function include_js_library($path)` where $path is the directory name of the asset under assets/js/ and the actual file name is index.js.

`function include_css_library($path)` is the same as the above, just requires the full file path.

`call_required_libraries($library_array)` allows multiple libraries to be called.  In your case, you would invoke it with `call_required_libraries('jquery-min-3-3-1','select2');`

The above requires renaming the files to index.js, and requires that the directory naming standard be put into the jquery-min-* naming standard.  Either a default case switch or a final elseif else statement needs to be added that invokes whatever path is passed in the argument array, OR add "select2" to the list of 'if' conditions.

These functions require some rework but when we change the functions themselves to make them more efficient (lots of people touched the file) and provide default vs. specified behavior of include_js_library($path) to be added to call_required_libraries , but we should not have to ever change the calls.

**Comment by Perseus** [comment]
@aethelwulffe I've moved the jquery file to the assets directory,  added a select2 clause to the headers file and added the required libraries function call to the edit list page

**Comment by aethelwulffe** [comment]
Yep, all the pieces work.

**Pull request merged**